[Core] Clear RuntimeOptions.featurePaths if rerun was used

tommywo opened the pull request:
<!-- These sections are meant as guidance for you. If something doesn't fit, you can just skip it. -->

## Summary
Fixes: #1630

## How Has This Been Tested?
Unit tests and a test run on my laptop

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

mpkorstanje commented on core/src/main/java/cucumber/runtime/RuntimeOptions.java:
```diff
@@ -69,6 +69,7 @@ public String map(String keyword) {
     private boolean wip = false;
     private SnippetType snippetType = SnippetType.UNDERSCORE;
     private int threads = 1;
+    private boolean isRerun = false;
```
I think this may need to be method scoped.

mpkorstanje commented:
Ah. Got a test case.

A cucumber annotation or CLI with a non-empty rerun file.

In combination with:

An environment variable without any features.

tommywo commented:
I've added a test case for that and added one more - if the is rerun file passed via cli and a feature it should ignore rerun file from cli -similar like it does to features.

mpkorstanje commented:
I'm having some trouble finding a test which uses:

CLI: `@file:path/rerun.txt`
ENVIRONMENT: `cucumber.options=--tags @smoke`

I think this will result in no feature files being selected to execute.

Could you point it out to me?

tommywo commented on core/src/main/java/cucumber/runtime/RuntimeOptions.java:
```diff
@@ -67,6 +67,7 @@ public String map(String keyword) {
     private boolean strict = false;
     private boolean monochrome = false;
     private boolean wip = false;
+    private boolean wasRerun = false;
```
right!

tommywo commented:
There was no such test with rerun file so i've added one. In that case behaviour is the same like passing features with line to cli - line numbers are ignored and only tags are used.

mpkorstanje commented:
All good now. Cheers!
